pkg/audit/reporter.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package audit
+
+import (
+	"context"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/provider"
+	dbx "storj.io/storj/pkg/statdb/dbx"
+	proto "storj.io/storj/pkg/statdb/proto"
+	"storj.io/storj/pkg/statdb/sdbclient"
+)
+
+type reporter interface {
+	RecordFailedAudits(ctx context.Context, failedNodes []*pb.Node) (err error)
+}
+
+// Reporter records audit reports in statdb and implements the reporter interface
+type Reporter struct {
+	statdb sdbclient.Client
+}
+
+var (
+	port   = ":7777"
+	apiKey = []byte("")
+)
+
+// NewReporter instantiates a reporter
+func NewReporter(ctx context.Context) (reporter *Reporter, err error) {
+	ca, err := provider.NewCA(ctx, 12, 14)
+	if err != nil {
+		return nil, err
+	}
+	identity, err := ca.NewIdentity()
+	if err != nil {
+		return nil, err
+	}
+	client, err := sdbclient.NewClient(identity, port, apiKey)
+	if err != nil {
+		return nil, err
+	}
+	return &Reporter{statdb: client}, nil
+}
+
+// RecordFailedAudits saves failed audit details to statdb
+func (reporter *Reporter) RecordFailedAudits(ctx context.Context, failedNodes []*pb.Node) (err error) {
+	nodes, err := reporter.recordEntryIfNotExists(ctx, failedNodes)
+	if err != nil {
+		return err
+	}
+
+	finished := false
+	retries := 0
+	for !finished && retries < 3 {
```

Comment:
You can get rid of `finished` and replace it with `for len(nodes) > 0 && retries < 3`